Update github workflows for better issue & PR management, automating chores

Wolfsblvt opened the pull request:
# Overview of Refactored Workflows

### 1. **Pull Request Automation**
#### `pr-auto-manager.yml`
**Purpose**: Centralizes all pull request labeling and comment automation.
Labeling PRs based on size and auto-applying relevant labels helps keeping PRs sorted, categorized and make them easy to search/review.

1. **Checkout** 
2. **Label on merge conflicts**
	- Automatically labels PRs when they have merge conflicts, and notifies the user (so it won't be so easy to miss this anymore)
3. **Label By Size**
	-  Auto-add the size labels again, now taking changed lines (added, modified, removed) into account
	- Posts a warning for PRs >1000 lines
4. **Label By Branch**  
   - Uses `pr-auto-labels-by-branch.yml` to apply labels based on branch name (e.g., _Bug Fix_, _New Feature_, _Against Release Branch_).  
   - Specifically label PRs against `release` with *❗ Against Release Branch*
   - Feature/bugfix based on common naming patterns (`fix/`, `fix-`, `feat/`, `feat-`,  `hotfix-`)
   - Targeted features based on parts of the branch name (`wi`, `mobile`, `tts`, `extension`, etc...)
5. **Label By Files**  
   - Uses `pr-auto-labels-by-files.yml` to apply labels (e.g., _Backend Changes_, _Build Changes_) if certain paths are touched.  
   - Current Labels:  
     - *🏭 Backend Changes*: `src/**`, `server.js`, etc.
     - *⚙️ config.yaml*: `default/config.yaml`
     - *🛠️ Build Changes*: `.github/workflows/**`, `docker/**` and more
     - *🌐 Language*: any changes in `public/locales/**`
     - *📥 Dependencies*: `package.json`, `public/lib/**`
6. **Directly Remove Stale on Activity**
	- This one is new. When a comment happens, all stale labels are removed, without the need of waiting for the nightly job.
7. **Add Merge Warnings on Labels**
	- When specific blocking labels have been set, merging will show a warning.
	- Includes: *🚫 Merge Conflicts*, *⛔ Don't Merge*, *🔨 Needs Work*, *🔬 Needs Testing*, *⛔ Waiting For External/Upstream*, *❗ Against Release Branch*, *💥💣 Breaking Changes*
8. **Post or Update Comments**  
   - Refers to `pr-auto-comments.yml` to post messages on the PR or even close it under certain label conditions, such as:  
     - *✖️ Invalid* → closes PR with an explanatory comment  
     - *💥💣 Breaking Changes* → warns about major releases  
     - *🔬 Needs Testing* → prompts contributors for testing
     - etc...

### 2. **Issue Automation**
#### `issues-auto-manager.yml`
**Purpose**: Automates how new or updated issues are labeled, commented on, or closed.
Easier to auto-label issues and adds some auto replies that make sense.

1. **Checkout**  
2. **Auto-Label Issues**  
   - References `issues-auto-labels.yml` to detect OS/platform or any other keywords (e.g., _Windows_, _Mac_, _Linux_).  
   - Automatically applies matching labels based on the issue text.
   - Added *🦊 Firefox* and *📱 Mobile* there
3. **Auto-Labels based on other Labels**
	- This one is new too. Adds/removes labels based on other labels, to keep it clean.
		- ✅ Add *👍 Approved* for relevant labels
		- ❌ Remove progress labels when issue is marked done or stale
		- ❌ Remove temporary labels when confirmed labels are added
		- ❌ Remove no bug labels when *🪲 Confirmed* is added
4. **Directly Remove Stale on Activity**
	- This one is new. When a comment happens, all stale labels are removed, without the need of waiting for the nightly job.
5. **Auto-Comment & Closure**  
   - Uses `issues-auto-comments.yml` to post messages or close the issue if certain labels (like *✖️ Invalid* or *❌ wontfix*) are applied.  
   - Locks the issue if labeled *💀 Spam*.

### 3. **Stale & Event Handlers**
#### `job-close-stale.yml`
**Purpose**: Manages stale issues and PRs by marking or closing them after inactivity.
Should help with nudging for responses and keeping the issues list a bit more clean

1. **Check Stale**  
   - Periodically run, applies *⚰️ Stale* after 6 months and later *🕸️ Inactive*, unless *📌 Keep Open*
2. **Check Awaiting Response**  
   - Periodically run, checks on *🚏 Awaiting User Response* to apply *🛑 No Response* after 7 days and later *🕸️ Inactive* and closes
   - This helps managing issues who require user input, but the user isn't active anymore.
3. **Check Alternative Exists**
	- Periodically run, checks on *🚧 Alternative Exists* to apply *🚏 Awaiting User Response* after 7 days and later *🕸️ Inactive* and closes
	- This nudges users to provide feedback if the alternative was enough to resolve the issues. Can be prevented by marking *📌 Keep Open*.

#### `on-open-handler.yml`
**Purpose**: Runs specific tasks when an issue or PR is opened.
Currently only exists to add some labels just on open.

1. **Label Maintainer Issues/PRs**
	- Straightforward. Checks the author against the people with repo permissions, and applies *👷 Maintainer*.

#### `on-close-handler.yml`
**Purpose**: Runs specific tasks when an issue or PR is closed.
Currently removes pending labels to keep the issue list cleaner for future searches.

##### Jobs & Key Steps
1. **Remove Pending Labels on Close**  
   - Removes any of the pending labels automatically
   - Labels: *🚏 Awaiting User Response*, *🧑‍💻 In Progress*, *📌 Keep Open*, *🚫 Merge Conflicts*, *🔬 Needs Testing*, *🔨 Needs Work*, *⚰️ Stale*, *⛔ Waiting For External/Upstream*

### Notes & Rationale

1. **Consolidated Config Files**  
   - Each workflow references unified YAML configs (e.g., `pr-auto-labels-by-files.yml`, `issues-auto-labels.yml`) to keep labeling rules in one place.
2. **Reduced Duplication**  
   - By combining logic into fewer workflows, we minimize overhead and improve maintainability.
3. **Consistent Contributor Experience**  
   - Labeling, comments, and stale checks are standardized. This helps contributors and maintainers quickly identify status and next steps.
4. **Easier Future Updates**  
   - Adjusting labels or rules is now as simple as editing a single `.yml` config file, rather than multiple scattered workflow files.


# Copilot
<details>
<summary>Copilot summary</summary>
This pull request includes significant updates to the GitHub workflows and issue/PR management configurations. The changes involve adding new labels, updating issue templates, and removing outdated workflows. The most important changes include the addition of new issue and PR label configurations, the removal of several workflows, and the introduction of new automated comment rules.

### Issue and PR Management Updates:
* [`.github/ISSUE_TEMPLATE/bug-report.yml`](diffhunk://#diff-7c49a365610d935ce439e3f241cd3f5d1df89a95d1a674d11ebdb1e518c7335cR2): Added `type: Bug` to the bug report template.
* [`.github/ISSUE_TEMPLATE/feature-request.yml`](diffhunk://#diff-a18ac139672d9f557f950ebe7df3f057ba19abe40d02ac2c81f72bcf44257b68R2): Added `type: Feature` to the feature request template.
* [`.github/issues-auto-comments.yml`](diffhunk://#diff-86d11df5bfd12fd67fa37717c6ca1b78c23829285b2a9669e084ea05c8ef7a2bR1-R69): Added new auto-comments for various issue labels such as `✖️ Invalid`, `👩‍💻 Good First Issue`, `❌ wontfix`, and more.
* [`.github/issues-auto-labels.yml`](diffhunk://#diff-dce2cffb4d190248e6d26cda7b8fae273b434c8c9e68bd59969eb9f4a55641afL1-L4): Renamed from `labeler.yml` and added new label configurations for `🦊 Firefox` and `📱 Mobile`. [[1]](diffhunk://#diff-dce2cffb4d190248e6d26cda7b8fae273b434c8c9e68bd59969eb9f4a55641afL1-L4) [[2]](diffhunk://#diff-dce2cffb4d190248e6d26cda7b8fae273b434c8c9e68bd59969eb9f4a55641afR15-R20)
* [`.github/pr-auto-comments.yml`](diffhunk://#diff-a699cacf48dcc3f436415ea8807ce5d72029cb2eddf1b2eff4629bebe948d4e1R1-R43): Introduced new auto-comments for PR labels like `✖️ Invalid`, `⛔ Don't Merge`, `💥💣 Breaking Changes`, and more.

### Workflow Removals:
* [`.github/workflows/add-comment-from-tag.yml`](diffhunk://#diff-e0b6fadb88a17b5bedc2ac30bdd3b13f08ba3022dc22e9dfb2bba3e45ed5e84dL1-L28): Removed workflow that added comments based on labels.
* [`.github/workflows/check-merge-conflicts.yml`](diffhunk://#diff-19dd7a85ae6dc45a7c07f5cab5902833540623c70ec1da7c4b55848e2fe53c0fL1-L17): Removed workflow that detected and labeled PRs with merge conflicts.
* [`.github/workflows/close-stale-issues.yml`](diffhunk://#diff-221ec56332617dc2f33423aaada2c915da7a8a49d88ad5189606cfa78518fea9L1-L82): Removed workflow that closed stale issues.
* [`.github/workflows/get-pr-size.yml`](diffhunk://#diff-01ba1e7b87c6dc805431a9f03769cb5a31bb61a2bb2e9520a8113d0ddb91c218L1-L39): Removed workflow that commented on PR size and labeled based on lines changed.

### New Workflows:
* [`.github/workflows/issues-auto-manager.yml`](diffhunk://#diff-36a474fa4807b6d0bf15cd404cdc0cd87e254ed630b57002ac6ceb555b75189cR1-R87): Added a new workflow to manage issue labels and post comments based on labels.

These updates aim to streamline the issue and PR management process, ensuring that the repository remains organized and contributors receive timely feedback.
</details>

Wolfsblvt committed:
Update PR size workflow

Wolfsblvt committed:
Update stale workflow

- Remove not needed "Maintainer Response" labeling
- Update to latest version
- Add link to source

Wolfsblvt committed:
Update remove labels on close workflow

- Added more labels that should be removed
- Included PRs and their labels to be auto-removed

Wolfsblvt committed:
Remove unused maintainers response workflow

Wolfsblvt committed:
Update merge conflicts workflow

- Change the action being used
- Add comment to notify PR author on merge conflicts

Wolfsblvt committed:
Refactor workflows into more structured files

Wolfsblvt committed:
"add Maintainer label" job via on open workflow

Wolfsblvt committed:
auto ask for feedback on "Alternative Exists"

Wolfsblvt committed:
Auto-comment/label issues once on staging

Wolfsblvt committed:
Auto-label PRs based on files + targeting release

Wolfsblvt committed:
Rewrite and expand auto comments

Wolfsblvt committed:
unstale issues/prs on comments right away

Wolfsblvt committed:
update labels based on other labels

Wolfsblvt committed:
Add check for PRs with merge-blocking labels

Wolfsblvt committed:
Add issue types to issue templates

Wolfsblvt committed:
Rework auto labels based on labels a bit

Wolfsblvt committed:
A bit more auto-labeling on issue open

Wolfsblvt committed:
Update auto comments to make them more uniform

Wolfsblvt committed:
PR workflow, add labels based on files/branch name

- Add PR labels based on changed files
- Add PR labels based on branch names

Wolfsblvt committed:
Directly remove stale labels on comments

Wolfsblvt committed:
Fix maintainer label, minor docs

Wolfsblvt committed:
Merge branch 'staging' into update-git-workflows

Wolfsblvt committed:
Rename some labels, add config label

- Rename PR size labels
- Add config.yaml label on config changes
- Add config.yaml to the backened changes label
- Awaiting User Response will not be blocked by Keep Open

Wolfsblvt committed:
Fix PR labeling failing

Wolfsblvt committed:
Move unstale to the issue/pr manager

Doesn't make sense to run the scheduled workflow and then skip most of it's jobs.
And it looks weird in the PR boxy thingy

Cohee1207 commented on .github/pr-auto-labels-by-branch.yml:
```diff
@@ -0,0 +1,77 @@
+- head-branch: ['^fix[/-]', '\bfixes\b']
+
+🚑 Hot Fix:
+- head-branch: ['^hotfix[/-]']
```
Does this mean we need to use these weird branch prefixes now?

Wolfsblvt replied:
No. It's totally optional. This is *mostly* for contributors, if they wanna have labels assigned right away.
Besides the weird ST words in there, it follows common branch naming schemas, like `feat/` or `fix/`, but... anyone can still name as they please.

Wolfsblvt committed:
Move PR size message to auto label replies

Wolfsblvt committed:
Extract workflows for merge conflicts + issue done

- Extract merge conflicts check to its own workflow, plus make it run on push
- Add issues update as Done or Done staging based on extracted commit messages
